### DISCUSSION TITLE
About the "NetMessage.SendData"

### DISCUSSION THREAD

**Discussion by IchiKen98** [Pryaxis/TShock#1729] [General]
Hello TShockers, i'm very new to programing so i have some questions about the "NetMessage.SendData".
1. What is "NetMessage.SendData"?
2. What is it used for? Some examples?
3. Does it need DLLs?

**Comment by hakusaro** [comment]
[NetMessage.SendData](https://github.com/Pryaxis/Sources/blob/master/1.3.5.3/Terraria/NetMessage.cs#L49) is basically the core messaging system in Terraria. TShock wraps it and does some other stuff with it, and it's generally more useable with our guidance than just trying to slam random calls into it. E.g., https://github.com/Pryaxis/TShock/blob/5fd6280bc8a39cdfb6f6d23e03a2150e2c8783c0/TShockAPI/TSPlayer.cs#L1128. It's used for essentially all server/client communication in one way or another.

To use it, you have to write your own TShock plugins or reverse engineer Terraria. Then, you can make as many calls to SendData as you want. You do need to be a C# programmer though.
